### PR TITLE
[RESTEASY-1854] Use proper default package to scan for JAX-RS application classes

### PR DESCRIPTION
**Pull request by geoand**
This not only improves startup time, but also avoids various problems
that can occur in tests due to classpath scanning

### CODE REVIEW

**Comment by geoand** [comment]
@asoldano I agree :) What version should I use?

**Comment by asoldano** [comment]
@geoand this would basically solve https://issues.jboss.org/browse/RESTEASY-1854, right?

**Comment by geoand** [comment]
@asoldano exactly

File: mds/USAGE.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ __Important notes__
...
-1. Starting on version 3.0.0, the behavior of the `scanning` JAX-RS Application subclass registration method has changed and has become more restrictive. Instead of scanning the whole classpath, it will scan only packages registered to be scanned by Spring framework (regardless of the JAX-RS Application subclass being a Spring bean or not). The reason is to improve application startup performance and avoiding classpath scanning issues in tests.
+1. Starting with version 2.1.0.Final, the behavior of the `scanning` JAX-RS Application subclass registration method has changed and has become more restrictive. Instead of scanning the whole classpath, it will scan only packages registered to be scanned by Spring framework (regardless of the JAX-RS Application subclass being a Spring bean or not). The reason is to improve application startup performance and avoiding classpath scanning issues in tests.
```
Comment (fabiocarvalho777) [inline mds/USAGE.md]: You are breaking semantic versioning by doing so. That is fine by the way, if that is what you want, but generally that is not a good idea, and I wouldn't recommend that. If you want to introduce a backward incompatible change, do so in a major version (and that is why I planned to do that in version 3.0.0 by the way, which I was reserving for Spring Boot 2, RESTEasy 4).

Comment (geoand) [reply mds/USAGE.md]: @fabiocarvalho777 I personally don't mind either way. Whatever you and @asoldano think is best, I'll obviously roll with :)

Comment (geoand) [reply mds/USAGE.md]: I will also follow up with another PR to bump the Spring Boot version to the latest SB 2.1.x after this one

Comment (asoldano) [reply mds/USAGE.md]: @geoand @fabiocarvalho777 I've actually been in doubt about the version, as besides this and the upgrade to Spring Boot 2.1 we won't have much in the new major and I believe @geoand will soon need a release. Having said this, the fact that @fabiocarvalho777 pointed out is enough to tell us that someone might be relying on the assumption that non backward compatible changes are only coming in majors, so if you don't mind @geoand please go for 3.0.0-SNAPSHOT. Thanks

**Comment by geoand** [comment]
@liweinan sounds good :)

**Pull request merged**